README.md

```diff
@@ -2,7 +2,7 @@
 
 [![Greenkeeper badge](https://badges.greenkeeper.io/Qard/breadth-filter.svg)](https://greenkeeper.io/)
 
-Apply a deep object filter via breadth traversal.
+Apply a deep object filter via breadth traversal. It allows replacing values, including retargeting objects and arrays and even detecting circular references to allow the user to decide how to handle the circle. Note that the filter will _not_ descend _into_ circular objects but will only _reach_ them.
```

Comment:
```suggestion
Apply a deep object filter via breadth traversal. It allows replacing values, including retargeting objects and arrays and even detecting circular references to allow the user to decide how to handle the circle.

By default, the filter will _not_ descend _into_ circular objects but will only _reach_ them - in essence filtering out those properties. This behavior can be overridden by supplying custom `onObject` and `onArray` functions.
```

Reply:
That's inaccurate. Even with a custom `onObject`, it will _not_ descend into an object it has already visited. The first visit will descend through it, but the second time will just _reach_ it, so you can decide what to put in that spot (ie: `[Circular]`)

Reply:
Ah, because the object have been added to `seen`. Got'ya 👍